File: 2020-09-12 - Complejidad algorítmica/0xf2 - Partir un arreglo en 2/linear.cpp
#include <vector>
#include <iostream>
using namespace std;


int main() {
  int n;
  cin >> n;
  int sum = 0;
  vector<int> a(n);
  for (int i = 0; i < n; ++i) {
    cin >> a[i];
    sum += a[i];
  }

  int left = 0;
  for (int i = 0; i < n; ++i) {
    int right = sum - left;
    if (left > 0 and right < 0) {
      cout << i << endl;
      return 0;
    }

    left += a[i];
  }
  cout << "Impossible" << endl;
  return 0;
}

File: 2020-09-12 - Complejidad algorítmica/0xf2 - Partir un arreglo en 2/linear.py
def solve(a):
  n = len(a)
  total = sum(a)

  left = 0
  for i in range(n):
    right = total - left
    if left > 0 and right < 0:
      print(i)
      return
    left += a[i]

  print("Impossible")

if __name__ == "__main__":
  n = int(input())
  a = [int(x) for x in input().split()]
  solve(a)